spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -179,6 +179,11 @@ sealed class SpdxExpression {
      * Concatenate [this][SpdxExpression] and [other] using [SpdxOperator.OR].
      */
     infix fun or(other: SpdxExpression) = SpdxCompoundExpression(this, SpdxOperator.OR, other)
+
+    /**
+     * Compare string representations of [SpdxExpression]s.
+     */
+    override fun compareTo(other: SpdxExpression) = toString().compareTo(other.toString())
```

Comment:
This implementation is not consistent with equal, e.g. `a AND b` and `b AND a` are equal but this function would return a value != 0. I think that's an issue because if we make `SpdxExpression` comparable it's likely to be used in sorted maps or sets.

Reply:
How do you propose to solve this? Should we make `compareTo()` consistent with `equals()`, or should we go with @fviernau's proposal to just provide a comparator (which might be inconsistent with `equals()`) which you need to explicitly use, or should we drop this change completely?

Reply:
I think I'd rather drop this change because I'm not sure if we really need sorting for SPDX expressions based on the semantics. Usually we want to sort the string representations, and for this it's better to make it explicit in code, and it's also easy to sort by properties in Kotlin.